gpii/node_modules/singleInstance/test/SingleInstanceTests.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * singleInstance Tests
+ *
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The R&D leading to these results received funding from the
+ * Department of Education - Grant H421A150005 (GPII-APCP). However,
+ * these results do not necessarily represent the policy of the
+ * Department of Education, and you should not assume endorsement by the
+ * Federal Government.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("universal"),
+    fs = require("fs"),
+    os = require("os"),
+    child_process = require("child_process");
+
+var jqUnit = fluid.require("node-jqunit");
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.tests.singleInstance");
+
+require("singleInstance");
+
+jqUnit.module("gpii.tests.singleInstance");
+
+jqUnit.test("Testing checkInstance, registerGpiiInstance, and deregisterInstance", function () {
+
+    var pidFile = os.tmpdir() + "/gpii-test-pid" + Date.now();
+
+    // Start another process (a cross-platform "sleep"), to test against a live pid.
+    var otherProcess = child_process.exec("node -e \"setTimeout(console.log, 5000)\"");
+
+    var tests = [
```

Comment:
It would be nicer to move all of this material out into the gpii.tests.singleInstance namespace we constructed - e.g. gpii.tests.singleInstance.testDefs for this and gpii.tests.singleInstance.deletePidFile for the function